src/EPR.Payment.Service/Services/Payments/PaymentsService.cs

```diff
@@ -0,0 +1,20 @@
+﻿using AutoMapper;
+using EPR.Payment.Service.Common.Data.Interfaces.Repositories.Payments;
+using EPR.Payment.Service.Services.Interfaces.Payments;
+
+namespace EPR.Payment.Service.Services.Payments
+{
+    public class PaymentsService : IPaymentsService
+    {
+        private readonly IPaymentsRepository _paymentsRepository;
+        public PaymentsService(IPaymentsRepository paymentsRepository)
+        {
+            _paymentsRepository = paymentsRepository ?? throw new ArgumentNullException(nameof(paymentsRepository));
+        }
+
+        public async Task<decimal> GetPreviousPaymentsByReferenceAsync(string reference, CancellationToken cancellationToken)
+        {
+            return await _paymentsRepository.GetPreviousPaymentsByReferenceAsync(reference, cancellationToken);
```

Comment:
I think we should check the value of reference here to make sure it is not null or empty.

Reply:
done